tests/test_telemetry.py

```diff
@@ -538,7 +539,7 @@ def test_app_started_product_info(self):
             if not is_v2_payload(data):
                 continue
             if get_request_type(data) == "app-started":
-                products = data["request"]["content"]["application"]["products"]
+                products = data["request"]["content"]["payload"]["products"]
```

Comment:
Products located inside `payload` not `application`: https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/auto-commit/GeneratedDocumentation/ApiDocs/v2/SchemaDocumentation/Schemas/app_started.md

Reply:
Well, it now fails on the modified line : https://github.com/DataDog/system-tests/actions/runs/6485987913/job/17613297835?pr=1546

`payload` does not have `products` key. Does it mean the python lib does not respect the RFC ?

Reply:
According to [the RFC](https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/main/GeneratedDocumentation/ApiDocs/v2/SchemaDocumentation/Schemas/app_started.md#app_started) `products` is not mandatory. Should we remove this assertion then? The current assertion is not correct because the `application` field is mandatory but according to [the RFC](https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/main/GeneratedDocumentation/ApiDocs/v2/SchemaDocumentation/Schemas/application.md#application) the `application` object does not have a `products` field.

Reply:
I also noticed [this example](https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/main/GeneratedDocumentation/ApiDocs/v2/SchemaDocumentation/Schemas/telemetry.md#example) in the RFC does not match [the `application` schema](https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/main/GeneratedDocumentation/ApiDocs/v2/SchemaDocumentation/Schemas/application.md#application)

Reply:
The test must follow the RFC :) But that's a good question. Does the RFc means "`products` is optional, whatever the context", or does it mean "`products` can be omitted if empty".

Because the test expect to see `appsec` in products.

If the answer is the first one, the test must be : `if products in payload then assert appsec in appsec in payload.products`.

Reply:
I've excluded Python in `test_app_started_product_info`

Reply:
The discrepancy was created when we updated our spec, moving the `products` payload from the `application` to `payload` in our v1 -> v2 upgrade. It looks like python hasn't reflected this change so this change looks fine to me. I will notify them of the change they need to make.